extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -31,20 +31,22 @@ import {
   getCorrelator,
 } from '../../../ads/google/a4a/utils';
 import {getLifecycleReporter} from '../../../ads/google/a4a/performance';
-import {documentStateFor} from '../../../src/document-state';
 import {getMode} from '../../../src/mode';
 import {stringHash32} from '../../../src/crypto';
 import {domFingerprintPlain} from '../../../src/utils/dom-fingerprint';
+import {viewerForDoc} from '../../../src/viewer';
 
 /** @const {string} */
 const ADSENSE_BASE_URL = 'https://googleads.g.doubleclick.net/pagead/ads';
 
-/** @const {!Object<string, string>} */
+/**
+ * See `VisibilityState` enum.
+ * @const {!Object<string, string>}
+ */
 const visibilityStateCodes = {
```

Comment:
Same question.  Returned visibility values that aren't in this mapping will get silently translated to 0 on the ad request call.  Not sure if that's acceptable.  Tagging @bobcassels , who knows much more about the ad requests than I do.